samples/go/poke/main.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/attic-labs/noms/go/dataset"
+	"github.com/attic-labs/noms/go/spec"
+	"github.com/attic-labs/noms/go/types"
+	flag "github.com/juju/gnuflag"
+)
+
+func main() {
+	var outDSStr = flag.String("out-ds-name", "", "output dataset to write to - if empty, defaults to input dataset")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Poke modifies a single value in a noms database.\n\n")
+		fmt.Fprintf(os.Stderr, "Usage: %s [-out-ds-name=<name>] <ds> <path> <new-val>\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  <ds>      : Dataset to modify\n")
+		fmt.Fprintf(os.Stderr, "  <path>    : Path to a value within <ds> to modify\n")
+		fmt.Fprintf(os.Stderr, "  <new-val> : new value for <path>\n\n")
+		fmt.Fprintln(os.Stderr, "Flags:\n")
+		flag.PrintDefaults()
+	}
+
+	flag.Parse(false)
+
+	if flag.NArg() == 0 {
+		flag.Usage()
+		return
```

Comment:
It would be good to get posix exit codes here. From what I've seen, the Go idiom is having a different function which returns an int, then calling `os.Exit` on it, like
```
func poke() (exitCode int) {
  if flag.Narg() == 0 {
    flag.Usage()
    return 1
  }
  ...
  return
}

func main() {
  os.Exit(poke())
}
```
E.g. https://github.com/golang/go/blob/master/src/cmd/gofmt/gofmt.go#L161 (that uses a global exitCode I suppose).